tests/perf/components/telemetry/main.go

```diff
@@ -0,0 +1,266 @@
+	"k8s.io/client-go/util/homedir"
+)
+
+var (
+	logPipelineGVR = schema.GroupVersionResource{Group: "telemetry.kyma-project.io", Version: "v1alpha1", Resource: "logpipelines"}
+	defaultTimeout = 15 * time.Minute
+)
+
+type flags struct {
+	count          int
+	kubeconfig     *string
+	timeout        time.Duration
+	host           string
+	port           int
+	uri            string
+	unhealthyRatio float64
+}
+
+type httpLogPipeline struct {
+	Name string
+	Tag  string
+	Host string
+	Port int
+	URI  string
+}
+
+func main() {
+	var f = flags{}
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig != "" {
+		f.kubeconfig = &kubeconfig
+	} else if home := homedir.HomeDir(); home != "" {
+		f.kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		f.kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+
+	flag.IntVar(&f.count, "count", 1, "Number of http log pipelines to deploy")
+	flag.DurationVar(&f.timeout, "timeout", defaultTimeout, "Timeout")
+	flag.StringVar(&f.host, "host", "", "Http host log pipelines will send logs to")
+	flag.StringVar(&f.uri, "uri", "", " URI Path")
+	flag.IntVar(&f.port, "port", 80, "Http port log pipelines will send logs to")
+	flag.Float64Var(&f.unhealthyRatio, "unhealthy-ratio", 0, "Ratio of unhealthy to healthy http log pipelines (from 0 to 1, where 0 means all are healthy and 1 means all are unhealthy)")
+	flag.Parse()
+
+	if err := run(f); err != nil {
+		fmt.Printf("Error: %v\n", err)
+		os.Exit(2)
+	}
+}
+
+func run(f flags) error {
+	if f.host == "" {
+		return errors.New("flag not specified: host")
+	}
+
+	if f.unhealthyRatio < 0 || f.unhealthyRatio > 1 {
+		return errors.New("flag invalid: unhealthyRatio (should be between 0 and 1)")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), f.timeout)
+	defer cancel()
+
+	config, err := clientcmd.BuildConfigFromFlags("", *f.kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	lokiPipelineYAML, err := os.ReadFile("./deploy/loki.yml")
+	if err := createLogPipeline(ctx, dynamicClient, lokiPipelineYAML); err != nil {
+		return err
+	}
+
+	g, gctx := errgroup.WithContext(ctx)
+
+	for i := 0; i < f.count; i++ {
+		current := i
+		g.Go(func() error {
+			var httpPipelineYAML []byte
+			unhealthy := int(f.unhealthyRatio * float64(f.count))
+			if current < unhealthy {
+				httpPipelineYAML, err = renderHTTPLogPipeline(f.host, "/bad", f.port, current)
+			} else {
+				httpPipelineYAML, err = renderHTTPLogPipeline(f.host, "/good", f.port, current)
+			}
+			if err != nil {
+				return err
+			}
+			if err := createLogPipeline(gctx, dynamicClient, httpPipelineYAML); err != nil {
+				return err
+			}
+			return nil
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return err
+	}
+
+	if err := collectMetrics(ctx, config); err != nil {
```

Comment:
Maybe we should have the pipelines running for a while before collecting metrics?

Reply:
Agree, added a `wait` flag to specify for how long to wait before collecting the metrics